### PR TITLE
20037 - Fix NSF unlock receipt

### PR DESCRIPTION
**Pull request by seeker25**
*Issue #:*
https://github.com/bcgov/entity/issues/20037

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: pay-api/src/pay_api/services/payment_account.py [hunk 1]
```diff
@@ -836,9 +837,7 @@ def create_account_event_payload(self, event_type: str, nsf_object: dict = None,
         }
 
         if event_type == MessageType.NSF_UNLOCK_ACCOUNT.value:
-            payload['data']['invoiceNumber'] = nsf_object['invoice_number']
-            payload['data']['paymentMethodDescription'] = nsf_object['payment_method_code']
-            payload['data']['receiptNumber'] = nsf_object['receipt_number']
+            payload['data'].update(receipt_info)
```
Comment (seeker25) [inline pay-api/src/pay_api/services/payment_account.py:840]: too risky for the future

**Comment by seeker25** [comment]
I'll push these both to DEV and test them out before merging

File: pay-api/src/pay_api/services/invoice.py [hunk 2]
```diff
@@ -379,10 +379,10 @@ def find_by_id(identifier: int, skip_auth_check: bool = False, one_of_roles=ALL_
         return invoice
 
     @staticmethod
-    def find_invoices_for_payment(payment_id: int) -> [Invoice]:
+    def find_invoices_for_payment(payment_id: int, status=InvoiceReferenceStatus.ACTIVE.value) -> List[Invoice]:
```
Comment (ochiu) [inline pay-api/src/pay_api/services/invoice.py:382]: Should be InvoiceStatus and not the reference status?

File: pay-api/src/pay_api/services/receipt.py [hunk 3]
```diff
@@ -200,3 +201,25 @@ def get_receipt_details(filing_data, invoice_identifier, skip_auth_check):
         # Format date to display in report.
         receipt_details['invoice']['createdOn'] = get_local_formatted_date(invoice_data.created_on)
         return receipt_details
+
+    @staticmethod
+    def get_nsf_receipt_details(payment_id):
+        """Return NSF receipt details, which can contain multiple invoices, combine these in PLI."""
+        receipt_details: dict = {}
+        invoices = Invoice.find_invoices_for_payment(payment_id, InvoiceReferenceStatus.COMPLETED.value)
```
Comment (ochiu) [inline pay-api/src/pay_api/services/receipt.py:209]: InvoiceStatus and not InvoiceReferenceStatus?

File: pay-api/src/pay_api/services/invoice.py (hunk 2 above)
Comment (ochiu) [reply pay-api/src/pay_api/services/invoice.py:382]: or maybe the parameter should be reference_status for clarity

File: pay-api/src/pay_api/services/receipt.py (hunk 3 above)
Comment (seeker25) [reply pay-api/src/pay_api/services/receipt.py:209]: needs to be invoice reference status, it was looking for ACTIVE by default before

**Pull request merged**